src/plugins/intel_cpu/src/nodes/transpose.cpp

```diff
@@ -37,8 +37,52 @@ bool Transpose::isSupportedOperation(const std::shared_ptr<const ov::Node>& op,
     return true;
 }
 
+class TransposeShapeInfer : public ShapeInferEmptyPads {
+public:
+    TransposeShapeInfer(const size_t& out_rank, const std::vector<size_t>& axes_vec) : m_out_rank(out_rank), m_axes_vec(axes_vec) {
+        m_outputShape = VectorDims(out_rank, 1);
+        m_needReverse = m_axes_vec.empty();
+    }
+
+    std::vector<VectorDims> infer(
+        const std::vector<std::reference_wrapper<const VectorDims>>& input_shapes,
+        const std::unordered_map<size_t, MemoryPtr>& data_dependency) override {
+        const VectorDims& shapeIn = input_shapes[0].get();
+        for (auto i = 0; i < m_out_rank; ++i) {
+            m_outputShape[i] = m_needReverse ? shapeIn[m_out_rank - 1 - i] : shapeIn[m_axes_vec[i]];
```

Comment:
> Consider re-factor to something like (to avoid check bool flag in every iteration):
> 
> ```c++
> if (m_needReverse){
>     // do reverse copy
> } else {
>     // do normal copy.
> }
> ```

Okay. Have re-factor it.